Lib/typing.py

```diff
@@ -174,11 +174,13 @@ def _type_check(arg, msg, is_argument=True, module=None, *, allow_special_forms=
     if (isinstance(arg, _GenericAlias) and
             arg.__origin__ in invalid_generic_forms):
         raise TypeError(f"{arg} is not valid as type argument")
-    if arg in (Any, NoReturn, ClassVar, Final):
+    if arg in (Any, NoReturn):
+        return arg
+    if allow_special_forms and arg in (ClassVar, Final):
         return arg
     if isinstance(arg, _SpecialForm) or arg in (Generic, Protocol):
         raise TypeError(f"Plain {arg} is not valid as type argument")
-    if isinstance(arg, (type, TypeVar, ForwardRef, types.UnionType, ParamSpec)):
+    if isinstance(arg, (type, TypeVar, ForwardRef, types.UnionType)):
```

Comment:
Will this affect putting ParamSpec inside Annotated? Looks like we don't have tests for that yet.

Reply:
I looked into this.

This check is needed in order to allow instances of `ParamSpec` (i.e. `P` in `P = ParamSpec("P")`) to pass `typing._type_check` because `callable(P)` is `False`. This can show up for instance like `Callable[Annotated[P,""], T]`.

Reply:
`Callable[Annotated[P, ""], T]` does not conform the PEP 612 specification.
```
callable ::= Callable "[" parameters_expression, type_expression "]"

parameters_expression ::=
  | "..."
  | "[" [ type_expression ("," type_expression)* ] "]"
  | parameter_specification_variable
  | concatenate "["
                   type_expression ("," type_expression)* ","
                   parameter_specification_variable
                "]"
```

Reply:
I'm not convinced it was the intent of the spec here to disallow Annotated, but you can also reach this code here:
```python
from typing import Callable, TypeVar, ParamSpec, get_type_hints
T = TypeVar("T")
P = ParamSpec("P")
def add_logging(f: Callable["P", T]):
    pass
get_type_hints(add_logging)
```

Reply:
Well, I re-added ParamSpec. Now some errors will not be caught at runtime. We will return to this in future.